File: platform/packages_py/env_resolver/env_resolver/env_statsig_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class StatsigEnv:
    api_key: str | None


def resolve_statsig_env(config: dict[str, Any] | None = None) -> StatsigEnv:
    return StatsigEnv(
        api_key=resolve(None, ['STATSIG_API_KEY', 'STATSIG_SERVER_SECRET'], config, 'api_key', None),
    )

File: platform/packages_mjs/env-resolver/src/env-statsig.x.ts
import { resolve } from '@internal/env-resolve';

export interface StatsigEnv {
    apiKey: string | undefined;
}

export function resolveStatsigEnv(config?: Record<string, any>): StatsigEnv {
    return {
        apiKey: resolve(undefined, ['STATSIG_API_KEY', 'STATSIG_SERVER_SECRET'], config, 'apiKey', undefined),
    };
}
